pydantic/json_schema.py

```diff
@@ -1210,6 +1124,80 @@ def render_warning_message(self, kind: JsonSchemaWarningKind, detail: str) -> st
         return f'{detail} [{kind}]'
 
 
+# ##### Start JSON Schema Generation Functions #####
+# TODO: These should be moved to the pydantic.funcs module or whatever when appropriate.
+
+
+def models_json_schema(
+    models: Sequence[type[BaseModel] | type[PydanticDataclass]],
+    *,
+    by_alias: bool = True,
+    title: str | None = None,
+    description: str | None = None,
+    ref_template: str = DEFAULT_REF_TEMPLATE,
+    schema_generator: type[GenerateJsonSchema] = GenerateJsonSchema,
+) -> dict[str, Any]:
+    # TODO: Put this in the "methods" module once that is created?
+    instance = schema_generator(by_alias=by_alias, ref_template=ref_template)
+    definitions = instance.generate_definitions([x.__pydantic_core_schema__ for x in models])
+
+    json_schema: dict[str, Any] = {}
+    if definitions:
+        json_schema['$defs'] = definitions
+    if title:
+        json_schema['title'] = title
+    if description:
+        json_schema['description'] = description
+
+    return json_schema
+
+
+# TODO: The JSON schema cache has been there from the beginning (https://github.com/pydantic/pydantic/pull/190 —
```

Comment:
let's move this into the PR body, I think it makes more sense there (at least in long form) than here.

Reply:
I can move the comment to the PR body, but let me know if you have thoughts about what you want to do with the cache.